v2/datastream-to-bigquery/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToBigQueryIT.java

```diff
@@ -0,0 +1,209 @@
+
+import static org.apache.beam.it.truthmatchers.PipelineAsserts.assertThatPipeline;
+import static org.apache.beam.it.truthmatchers.PipelineAsserts.assertThatResult;
+
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.datastream.v1.DestinationConfig;
+import com.google.cloud.datastream.v1.SourceConfig;
+import com.google.cloud.datastream.v1.Stream;
+import com.google.cloud.teleport.metadata.TemplateIntegrationTest;
+import com.google.pubsub.v1.SubscriptionName;
+import com.google.pubsub.v1.TopicName;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import org.apache.beam.it.common.PipelineLauncher;
+import org.apache.beam.it.common.PipelineLauncher.LaunchConfig;
+import org.apache.beam.it.common.PipelineOperator;
+import org.apache.beam.it.common.utils.PipelineUtils;
+import org.apache.beam.it.common.utils.ResourceManagerUtils;
+import org.apache.beam.it.gcp.TemplateTestBase;
+import org.apache.beam.it.gcp.bigquery.BigQueryResourceManager;
+import org.apache.beam.it.gcp.bigquery.conditions.BigQueryRowsCheck;
+import org.apache.beam.it.gcp.datastream.DatastreamResourceManager;
+import org.apache.beam.it.gcp.datastream.MySQLSource;
+import org.apache.beam.it.gcp.pubsub.PubsubResourceManager;
+import org.apache.beam.it.jdbc.CustomMySQLResourceManager;
+import org.apache.beam.it.jdbc.JDBCResourceManager;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.testcontainers.shaded.org.apache.commons.lang3.RandomStringUtils;
+
+@TemplateIntegrationTest(DataStreamToBigQuery.class)
+@RunWith(JUnit4.class)
+public class DataStreamToBigQueryIT extends TemplateTestBase {
+  private CustomMySQLResourceManager jdbcResourceManager;
+  private PubsubResourceManager pubsubResourceManager;
+  private DatastreamResourceManager datastreamResourceManager;
+  private BigQueryResourceManager bigQueryResourceManager;
+
+  private static final int ROW_COUNT = 10;
+
+  private static final String ID = "id";
+  private static final String JOB = "job";
+  private static final String NAME = "name";
+
+  @Before
+  public void setUp() throws IOException {
+    pubsubResourceManager =
+        PubsubResourceManager.builder(testName, PROJECT, credentialsProvider).build();
+    bigQueryResourceManager =
+        BigQueryResourceManager.builder(testName, PROJECT, credentials).build();
+    datastreamResourceManager =
+        DatastreamResourceManager.builder(testName, PROJECT, REGION)
+            .setCredentialsProvider(credentialsProvider)
+            .setPrivateConnectivity("datastream-private-connect-us-central1")
+            .build();
```

Comment:
Hard-coding connection details makes it harder for others to run IT's. Could you add it as a TestProperty:
```suggestion
    datastreamResourceManagerBuilder =
        DatastreamResourceManager.builder(testName, PROJECT, REGION)
            .setCredentialsProvider(credentialsProvider)
    String privateConnectivity = System.getProperty("privateConnectivity")
    if (privateConnectivity != null) {
      datastreamResourceManagerBuilder
              .setPrivateConnectivity(privateConnectivity)
    }
    datastreamResourceManager = datastreamResourceManagerBuilder.build();
```
And then add the flag to the gitactions framework: https://github.com/GoogleCloudPlatform/DataflowTemplates/compare/main...Polber:DataflowTemplates:jkinard/private-connectivity?expand=1